app/src/main/java/ch/hikemate/app/ui/map/MapScreen.kt

```diff
@@ -45,19 +46,80 @@ import androidx.compose.ui.unit.dp
 import androidx.compose.ui.viewinterop.AndroidView
 import androidx.lifecycle.viewmodel.compose.viewModel
 import ch.hikemate.app.R
+import ch.hikemate.app.model.route.Bounds
+import ch.hikemate.app.model.route.HikeRoute
+import ch.hikemate.app.model.route.LatLong
 import ch.hikemate.app.model.route.ListOfHikeRoutesViewModel
 import org.osmdroid.config.Configuration
 import org.osmdroid.events.MapListener
 import org.osmdroid.events.ScrollEvent
 import org.osmdroid.events.ZoomEvent
 import org.osmdroid.util.GeoPoint
 import org.osmdroid.views.MapView
+import org.osmdroid.views.overlay.Polyline
 
 object MapScreen {
   const val TEST_TAG_MAP = "map"
   const val TEST_TAG_MENU_BUTTON = "menuButton"
   const val TEST_TAG_HIKES_LIST = "hikesList"
   const val TEST_TAG_HIKE_ITEM = "hikeItem"
+
+  const val MIN_HUE = 0
+  const val MAX_HUE = 360
+  const val MIN_SATURATION = 42
+  const val MAX_SATURATION = 98
+  const val MIN_LIGHTNESS = 40
+  const val MAX_LIGHTNESS = 90
+
+  const val STROKE_WIDTH = 10f
+}
+
+/**
+ * Generates a random color in the HSL color space. This function uses the HSL color space to
+ * generate a random color as the HSL is better to generate more visually appealing colors.
+ *
+ * @return The generated color as an [Int].
+ */
+fun getRandomColor(): Int {
+  fun randomInt(min: Int, max: Int): Int {
+    return (min..max).random()
+  }
+
+  val h = randomInt(MapScreen.MIN_HUE, MapScreen.MAX_HUE).toFloat() // All colors
+  val s =
+      randomInt(MapScreen.MIN_SATURATION, MapScreen.MAX_SATURATION) /
+          100.0f // Saturation between 42% and 98%
+  val l =
+      randomInt(MapScreen.MIN_LIGHTNESS, MapScreen.MAX_LIGHTNESS) /
+          100.0f // Lightness between 40% and 90%
+
+  return Color.hsl(h, s, l).toArgb()
+}
+
+/**
+ * Shows a hike on the map.
+ *
+ * @param mapView The map view where the hike will be shown.
+ * @param hike The hike to be shown.
+ * @param color The color of the hike.
+ */
+fun showHikeOnMap(mapView: MapView, hike: HikeRoute, color: Int) {
+  val line = Polyline()
+
+  line.setPoints(hike.ways.map { GeoPoint(it.lat, it.lon) })
+  line.outlinePaint.color = color
+  line.outlinePaint.strokeWidth = MapScreen.STROKE_WIDTH
+
+  line.setOnClickListener { _, _, _ ->
+    Toast.makeText(
+            mapView.context,
+            "Hike details not implemented yet. Hike ID: ${hike.id}",
+            Toast.LENGTH_SHORT)
+        .show()
+    true
+  }
```

Comment:
Praise: very elegant method, I like it !

Just to mention, I added this `setOnClickListener` to the backlog so that it gets updated when we implement the Hike details screen.